src/newfight.cpp

```diff
@@ -1360,3 +1411,43 @@ void engage_close_combat_if_appropriate(struct combat_data *att, struct combat_d
     }
   }
 }
+
+bool handle_flame_aura(struct combat_data *att, struct combat_data *def) {
+  // Now that we've counterstrike-swapped if applicable, check and try to apply flame aura.
+  if (AFF_FLAGGED(def->ch, AFF_FLAME_AURA) || MOB_FLAGGED(def->ch, MOB_FLAMEAURA)) {
+    int force;
+
+    // First, we require that the attacker really is fighting in melee and unarmed. Flame aura only burns body parts.
+    if (att->ranged_combat_mode || !att->weapon) {
+      // Flameaura-flagged NPCs just use their own level as the force.
+      if (MOB_FLAGGED(def->ch, MOB_FLAMEAURA)) {
+        force = GET_LEVEL(def->ch);
+      }
+      else {
+        // Iterate through their spells, find the flame aura that's applied to them, and extract its force.
+        for (struct sustain_data *sust = GET_SUSTAINED(def->ch); sust; sust = sust->next) {
+          if (!sust->caster && sust->spell == SPELL_FLAMEAURA) {
+            force = sust->force;
+            break;
+          }
+        }
+      }
+
+      if (GET_EQ(att->ch, WEAR_HANDS) && GET_EQ(att->ch, WEAR_FEET)) {
+        // TODO: Apply impact armor.
+      }
+
+      // TODO: Now that we've set the force, deal (Force)M damage to the attacker. Use TYPE_SUFFERING maybe?
+      // Remember that damage() returns true if it's killed them-- if this happens, you'll need to abort handling, because att->ch is already nulled out.
+      if (damage(att->ch, ...)) {
+        // TODO: send a message to the defender and the room. Remember not to use att->ch here since it's been nulled by death.
```

Comment:
Cool message here too! Just send it to the defender and to the room instead of directly to the attacker, since the attacker at this point is dead (RIP).